ethportal-api/src/types/execution/header.rs

```diff
@@ -557,6 +605,109 @@ mod tests {
+        let decoded: Header = rlp::decode(&data).unwrap();
+        let expected: Header = Header {
+            parent_hash: H256::from_str(
+                "3a9b485972e7353edd9152712492f0c58d89ef80623686b6bf947a4a6dce6cb6",
+            )
+            .unwrap(),
+            uncles_hash: H256::from_str(
+                "1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
+            )
+            .unwrap(),
+            author: H160::from_str("2adc25665018aa1fe0e6bc666dac8fc2697ff9ba").unwrap(),
+            state_root: H256::from_str(
+                "3c837fc158e3e93eafcaf2e658a02f5d8f99abc9f1c4c66cdea96c0ca26406ae",
+            )
+            .unwrap(),
+            transactions_root: H256::from_str(
+                "4409cc4b699384ba5f8248d92b784713610c5ff9c1de51e9239da0dac76de9ce",
+            )
+            .unwrap(),
+            receipts_root: H256::from_str(
+                "46cab26abf1047b5b119ecc2dda1296b071766c8b1307e1381fcecc90d513d86",
+            )
+            .unwrap(),
+            logs_bloom: Bloom::default(),
+            difficulty: U256::from_str("0").unwrap(),
+            number: 0x1,
+            gas_limit: U256::from_str("0x7fffffffffffffff").unwrap(),
+            gas_used: U256::from_str("0x02a865").unwrap(),
+            timestamp: 0x079e,
+            extra_data: vec![0x42],
+            mix_hash: Some(
+                H256::from_str("56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421")
+                    .unwrap(),
+            ),
+            nonce: Some(H64::zero()),
+            base_fee_per_gas: Some(U256::from_str("9").unwrap()),
+            withdrawals_root: Some(
+                H256::from_str("56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421")
+                    .unwrap(),
+            ),
+            blob_gas_used: Some(U64::from_str("020000").unwrap()),
+            excess_blob_gas: Some(U64::from_str("0").unwrap()),
+            parent_beacon_block_root: None,
+        };
+        assert_eq!(decoded, expected);
+
+        let expected_hash =
+            H256::from_str("0x10aca3ebb4cf6ddd9e945a5db19385f9c105ede7374380c50d56384c3d233785")
+                .unwrap();
+        assert_eq!(decoded.hash(), expected_hash);
+    }
+
+    #[test]
+    fn post_dencun_header_without_blob_txs() {
+        let body =
+            std::fs::read_to_string("../test_assets/mainnet/block_19433902_value.json").unwrap();
+        let response: Value = serde_json::from_str(&body).unwrap();
+        let header: Header = serde_json::from_value(response["result"].clone()).unwrap();
+        let expected_hash =
+            H256::from_slice(&hex_decode(response["result"]["hash"].as_str().unwrap()).unwrap());
+        assert_eq!(header.number, 19433902);
+        assert_eq!(header.hash(), expected_hash);
+    }
+
+    #[test]
```

Comment:
I think you can also just use `rstest` here for the two `post_dencun_header_*` tests